### PR TITLE
feat(page): support dropping into blocks

### PR DESCRIPTION
**Pull request by wfnuser**
Presenting an initial PR aimed at addressing issue #3448 . As I'm currently acquainting myself with the codebase, there could be several issues that need attention. I kindly request you to review it and provide guidance. I plan to add more details in the description and refine the commit message in the near future.

### CODE REVIEW

**Comment by wfnuser** [comment]
I've noticed a quiver in the indicator when dragging blocks. I'm currently unsure about the solution. Additionally, I'm uncertain about how to add supplementary tests for this issue. Is there a method to simulate the dragging behavior for testing purposes?

File: packages/blocks/src/__internal__/utils/drag-and-drop.ts [hunk 1]
```diff
@@ -16,7 +17,11 @@ import { type EditingState } from './types.js';
 /**
  * A dropping type.
  */
-export type DroppingType = 'none' | 'before' | 'after' | 'database';
+/**
+ * Question: Shall we add some comments here to describe each dropping type's meaning.
+ * I'd like to add a new type for drop a block into another.
+ */
+export type DroppingType = 'none' | 'before' | 'after' | 'database' | 'in';
```
Comment (donteatfriedrice) [inline packages/blocks/src/__internal__/utils/drag-and-drop.ts:24]: Sure, we can add some comments for these types. I think 'before' means drop blocks before target block, 'after' means drop blocks after target block, 'database' means drop blocks in database.

**Comment by donteatfriedrice** [comment]
> I've noticed a quiver in the indicator when dragging blocks. I'm currently unsure about the solution. Additionally, I'm uncertain about how to add supplementary tests for this issue. Is there a method to simulate the dragging behavior for testing purposes?

Hello, we are glad that you have interest to make some contributions, thanks for your work. What do you mean 'a quiver in the indicator when dragging blocks', could you upload a video or GIF so that maybe we can solve it together. As for the test, you can check out [drag handle tests](https://github.com/toeverything/blocksuite/blob/c3aa109d6e455727aebb0b2048e8bd8158592399/tests/drag.spec.ts#L673) or other tests in tests/drag.spec.ts.

**Comment by wfnuser** [comment]
> Hello, we are glad that you have interest to make some contributions, thanks for your work. What do you mean 'a quiver in the indicator when dragging blocks', could you upload a video or GIF so that maybe we can solve it together. As for the test, you can check out [drag handle tests](https://github.com/toeverything/blocksuite/blob/c3aa109d6e455727aebb0b2048e8bd8158592399/tests/drag.spec.ts#L673) or other tests in tests/drag.spec.ts.

Certainly. I'd like to spend some time understanding how tests function in blocksuite first. For functions like await page.mouse.down();, is this a method for emulating real interactive actions in the tests? Is there some recommended reference links for me to get familiar with this tech stack.

Regarding the quiver issue, it appears as shown below, but I'm uncertain if this is expected behavior. (The right edge of indicator seems not so right to me.)

https://github.com/toeverything/blocksuite/assets/8191686/7104eca6-4525-46dc-b0ef-7c2b781ae4df

File: packages/blocks/src/__internal__/utils/drag-and-drop.ts (hunk 1 above)
Comment (wfnuser) [reply packages/blocks/src/__internal__/utils/drag-and-drop.ts:24]: So what's the difference between drop in database and not drop in database? I thought all the data are saved in database?

**Comment by donteatfriedrice** [comment]
> > Hello, we are glad that you have interest to make some contributions, thanks for your work. What do you mean 'a quiver in the indicator when dragging blocks', could you upload a video or GIF so that maybe we can solve it together. As for the test, you can check out [drag handle tests](https://github.com/toeverything/blocksuite/blob/c3aa109d6e455727aebb0b2048e8bd8158592399/tests/drag.spec.ts#L673) or other tests in tests/drag.spec.ts.
> 
> Certainly. I'd like to spend some time understanding how tests function in blocksuite first. For functions like await page.mouse.down();, is this a method for emulating real interactive actions in the tests? Is there some recommended reference links for me to get familiar with this tech stack.
> 
> Regarding the quiver issue, it appears as shown below, but I'm uncertain if this is expected behavior. (The right edge of indicator seems not so right to me.)
> 
>  Untitled.mov 
> And I noticed that there is still a bug in my implementation I will fix it soon.

Maybe you can get familiar with PlayWright by reading its docs. https://playwright.dev/docs/api/class-test

**Review by doodlewind — COMMENTED** [review]
@wfnuser seems that there are more details to be polished, do you have recent plan working on this PR?

**Comment by wfnuser** [comment]
> @wfnuser seems that there are more details to be polished, do you have recent plan working on this PR?

Yep. Something has happened recently, if it's not in a hurry can I handle it later (like next week). I found the test framework still need some time to explore, but I have to deal with other issues first.😂

**Comment by wfnuser** [comment]
Since the upstream has been changed a bit, I will open a new PR later.

**Comment by wfnuser** [comment]
@doodlewind @donteatfriedrice  I have solve the conflict and start looking at the test failures. I think there are some of the failures caused by the new drop in feature. If you do `dragHandleFromBlockToBlockBottomById` now, it will drop in the blocks not append it after the target block.
How should I deal with it, change the `dragHandleFromBlockToBlockBottomById` implementation to make it behave like append not drop in? I'm not sure about it. Maybe I need some help. If someone can pair code with me is much better.

BTW, I found that I don't have the access to reopen this pr. Can someone do it for me. Thanks.

**Comment by doodlewind** [comment]
@wfnuser seems there are some git related issue so I'm also unable to reopen it.

**Comment by wfnuser** [comment]
> @wfnuser seems there are some git related issue so I'm also unable to reopen it.

I reopen a new pr, and might need some help.

**Pull request closed without merging**